packages/core/test/acceptance/styling_spec.ts

```diff
@@ -2627,25 +2624,383 @@ describe('styling', () => {
     expect(getComputedStyle(div).width).toBe('10px');
   });
 
-  it('should allow classes with trailing and leading spaces in [ngClass]', () => {
+  it('should allow multiple styling bindings to work alongside property/attribute bindings', () => {
     @Component({
       template: `
-        <div leading-space [ngClass]="{' foo': applyClasses}"></div>
-        <div trailing-space [ngClass]="{'foo ': applyClasses}"></div>
+        <div 
+            dir-that-sets-styles
+            [style]="{'font-size': '300px'}"
+            [attr.title]="'my-title'"
+            [attr.data-foo]="'my-foo'">
+        </div>`
+    })
+    class MyComp {
+    }
+
+    @Directive({selector: '[dir-that-sets-styles]'})
+    class DirThatSetsStyling {
+      @HostBinding('style.width') public w = '100px';
+      @HostBinding('style.height') public h = '200px';
+    }
+
+    const fixture = TestBed.configureTestingModule({declarations: [MyComp, DirThatSetsStyling]})
+                        .createComponent(MyComp);
+    fixture.detectChanges();
+    const div = fixture.nativeElement.querySelector('div') !;
+    expect(div.style.getPropertyValue('width')).toEqual('100px');
+    expect(div.style.getPropertyValue('height')).toEqual('200px');
+    expect(div.style.getPropertyValue('font-size')).toEqual('300px');
+    expect(div.getAttribute('title')).toEqual('my-title');
+    expect(div.getAttribute('data-foo')).toEqual('my-foo');
+  });
+
+  it('should allow host styling on the root element with external styling', () => {
+    @Component({template: '...'})
+    class MyComp {
+      @HostBinding('class') public classes = '';
+    }
+
+    const fixture =
+        TestBed.configureTestingModule({declarations: [MyComp]}).createComponent(MyComp);
+    fixture.detectChanges();
+    const root = fixture.nativeElement as HTMLElement;
+    expect(root.className).toEqual('');
+
+    fixture.componentInstance.classes = '1 2 3';
+    fixture.detectChanges();
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('1 2 3');
+
+    root.classList.add('0');
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('0 1 2 3');
+
+    // TODO(pk): why ???
+    fixture.componentInstance.classes = '1 2 3 4';
+    fixture.detectChanges();
+    expect(root.className.split(/\s+/).sort().join(' ')).toEqual('0 1 2 3 4');
```

Comment:
This assert fails in VE with `Error: Expected '1 2 3 4' to equal '0 1 2 3 4'.`. Is is a deliberate breaking change?